### PR TITLE
Test PermutationImportance with XGBClassifier and pd.DataFrame issue#256

### PR DESCRIPTION
**Pull request by harrysalmon**
`PermutationImportance.fit` fails with XGBClassifier when passed a pandas Dataframe but fits when passed an numpy array. 

A test for the error in [#256](https://github.com/TeamHG-Memex/eli5/issues/256) @ianozsvald

May also be related to general xgb >= 0.7 issue [#259](https://github.com/TeamHG-Memex/eli5/issues/259)

### CODE REVIEW

**Comment by ianozsvald** [comment]
Many thanks for the test @harrysalmon . To the core devs on ELI5 - this test is a result of a "Make your firsts open source contribution" that I co-ran at PyDataLondon 2018 yesterday. Cheers!

**Comment by lopuhin** [comment]
Great, thanks a lot for contributing the test @harrysalmon  and for finding the issue @ianozsvald !
